subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ValidateStep.java

```diff
@@ -92,5 +104,9 @@ protected void recordProblem(Severity severity, String message) {
         public ImmutableMultimap<Severity, String> getProblems() {
             return problems.build();
         }
+
+        public ImmutableList<Class<?>> getTypes() {
```

Comment:
Why do we allow multiple types here? I don't see a place where we would be able to report multiple types. If there is, then there should be a test that we see the warning with multiple type names.

Reply:
I added a test in `ValidateStepTest` to check this.

Reply:
I understand that this can happen when `createContext` is called twice. Though it doesn't seem we do this anywhere, do we?

Reply:
We don't do it now, but we'll probably going to use it when validation happens for artifact transforms via this thing. Another thing that could be declared across multiple types is tasks with their additional actions.

Another alternative would be to introduce something like `UnitOfWork.getRootType()` or something, but I wanted to avoid doing that, because it would prevent multiple types to be associated with a single unit of work.